src/doc/rustdoc/src/passes.md

```diff
@@ -1,3 +1,143 @@
 # Passes
 
-Coming soon!
\ No newline at end of file
+Rustdoc has a concept called "passes". These are transformations that
+`rustdoc` runs on your documentation before producing its final output.
+
+In addition to the passes below, check out the docs for these flags:
+
+* [`--passes`](command-line-arguments.html#--passes-add-more-rustdoc-passes)
+* [`--no-defaults`](ommand-line-arguments.html#--no-defaults-dont-run-default-passes)
+
+## Default passes
+
+By default, rustdoc will run some passes, namely:
+
+* `strip-hidden`
+* `strip-private`
+* `collapse-docs`
+* `unindent-comments`
+
+However, `strip-private` implies `strip-private-imports`, and so effectively,
+all passes are run by default.
+
+## `strip-hidden`
+
+This pass implements the `#[doc(hidden)]` attribute. When this pass runs, it
+checks each item, and if it is annotated with this attribute, it removes it
+from `rustdoc`'s output.
+
+Without this pass, these items will remain in the output.
+
+## `unindent-comments`
+
+When you write a doc comment like this:
+
+```rust,ignore
+/// This is a documentation comment.
+```
+
+There's a space between the `///` and that `T`. That spacing isn't intended
+to be a part of the output; it's there for humans, to help separate the doc
+comment syntax from the text of the comment. This pass is what removes that
+space.
+
+It takes this indentation amount from the first line. For example:
+
+```text
+    line1
+    line2
+```
+
+becomes
+
+```text
+line1
+line2
+```
+
+In this case, both lines have the same indentation. But if the second one is
+indented more than the first, it respects it:
+
+```text
+  line1
+      line2
+```
+
+it will become
+
+```text
+line1
+    line2
```

Comment:
Actually, this one would also be:

````markdown
```text
line1
line2
```
````

However

````markdown
```text
  line1

      line2
```
````

becomes:

````markdown
```text
line1

    line2
```
````

Whether the second line is entirely white-space or not makes a difference.